snuba/datasets/groups.py

```diff
@@ -0,0 +1,124 @@
+import re
+
+from datetime import timedelta
+from typing import Mapping, Sequence
+
+from snuba.datasets import TimeSeriesDataset
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.factory import get_dataset
+from snuba.datasets.schemas.join import (
+    JoinConditionExpression,
+    JoinCondition,
+    JoinedSchema,
+    JoinStructure,
+    JoinType,
+    SchemaJoinedSource,
+)
+from snuba.query.extensions import (
+    PerformanceExtension,
+    ProjectExtension,
+    QueryExtension,
+)
+from snuba.query.timeseries import TimeSeriesExtension
+
+
+class Groups(TimeSeriesDataset):
+    """
+    Experimental dataset that provides Groups data joined with
+    the events table.
+    """
+
+    EVENTS_ALIAS = "events"
+    GROUPS_ALIAS = "groups"
+
+    QUALIFIED_COLUMN_REGEX = re.compile(r"^([a-zA-Z_][a-zA-Z0-9_]*)\.([a-zA-Z0-9_\.\[\]]+)$")
+
+    def __init__(self) -> None:
+        self.__grouped_message = get_dataset("groupedmessage")
+        self.__events = get_dataset("events")
+
+        join_structure = JoinStructure(
+            left_source=SchemaJoinedSource(
+                self.GROUPS_ALIAS,
+                self.__grouped_message.get_dataset_schemas().get_read_schema(),
+            ),
+            right_source=SchemaJoinedSource(
+                self.EVENTS_ALIAS,
+                self.__events.get_dataset_schemas().get_read_schema(),
+            ),
+            mapping=[
+                JoinCondition(
+                    left=JoinConditionExpression(
+                        table_alias=self.GROUPS_ALIAS,
+                        column="project_id"),
+                    right=JoinConditionExpression(
+                        table_alias=self.EVENTS_ALIAS,
+                        column="project_id"),
+                ),
+                JoinCondition(
+                    left=JoinConditionExpression(
+                        table_alias=self.GROUPS_ALIAS,
+                        column="id"),
+                    right=JoinConditionExpression(
+                        table_alias=self.EVENTS_ALIAS,
+                        column="group_id"),
+                ),
+            ],
+            join_type=JoinType.LEFT,
+        )
+
+        schema = JoinedSchema(join_structure)
+        dataset_schemas = DatasetSchemas(
+            read_schema=schema,
+            write_schema=None,
+        )
+        super().__init__(
+            dataset_schemas=dataset_schemas,
+            time_group_columns={
+                'events.time': 'events.timestamp',
+            },
+        )
+
+    def default_conditions(self):
+        return [
+            ('events.deleted', '=', 0),
+            ('groups.record_deleted', '=', 0),
+        ]
```

Comment:
This seems like it'll be a little brittle, hopefully this is temporary.

Reply:
It's from before I supported table_alias in the datasets, then I missed it. I can remove it right away.